core/src/main/java/org/apache/iceberg/variants/VariantMetadata.java

```diff
@@ -34,4 +34,20 @@ public interface VariantMetadata extends Variants.Serialized {
 
   /** Returns the size of the metadata dictionary. */
   int dictionarySize();
+
+  static String asString(VariantMetadata metadata) {
+    StringBuilder builder = new StringBuilder();
```

Comment:
Maybe just add as regular method rather than static function. 

That also will be consistent with other conversion like as asStructType() in Type.java.

Reply:
This can't be added in the interface because it would be a default that conflicts with an existing method (`Object#toString`). To share the implementations, I added these as static methods in the interfaces and then call them from the concrete classes.

Reply:
Yeah. We need to add as default. We are adding `asString()`, not `toString()`.  Why will that conflict? I tried and seems that works.

Reply:
The only reason this is added as `asString` is that it is a different signature. That is the static method used to convert.